server.js

```diff
@@ -72,8 +72,9 @@ app.get('/', function(req, res) {
 
 /* check username and password and send appropriate response */
 app.post('/login', function(req, res) {
-    if (req.session.user) {
+    if ('user' in req.session) {
         req.session.destroy();
+        return res.status(400).send('Invalid Request');
```

Comment:
what does this mean? so confused

Reply:
This basically means the user tried to log in again when it was already logged in. So I just kill it for now.

Reply:
how does it break it then. If they already logged in then we kill their previous session and open the door for a new session

Reply:
After you destroy the session, you need to return back an error msg. otherwise it goes down the code and trys to add req.session.user = null, and it breaks. cause we just destroyed it.

Reply:
then remove the req.session.user = null as its pointless.

Reply:
then too you still need:
if('user' in req.session):
    destroy...
    return statement

Reply:
yes we still need that